### PR TITLE
docs(storage-resize-image): update the description for Sharp options param

### LINKED ISSUE #1891: Scale Image Instead of Cropping?
Do not see the configuration to allow resize with a preserved aspect ratio.

<div type='discussions-op-text'>

<sup>Originally posted by **rstarr-indeed** December  5, 2023</sup>
It looks like the extension is cropping images instead of scaling them like it's supposed to. Is there an option I need to set to change this?

image
</div>

### PR DESCRIPTION
**Pull request by renuvkelkar**
fixes #1891

### CODE REVIEW

**Review by cabljac — APPROVED** [review]
looks good to me!

File: storage-resize-images/README.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ Leave this field empty if you do not want to store failed images in a separate d
 * Convert image to preferred types: The image types you'd like your source image to convert to.  The default for this option will be to keep the original file type as the destination file type.
 
 
-* Output options for selected formats: Provide a optional output option stringified object containing Sharp Output Options for selected image types conversion. eg. `{"jpeg": { "quality": 5, "chromaSubsampling": "4:4:4" }, "png": { "pallete": true }}`
+* Output options for selected formats: Provide a optional output option stringified object containing Sharp Output Options for selected image types conversion. eg. {"jpeg": { "quality": 5, "chromaSubsampling": "4:4:4" }, "png": { "pallete": true }} and {"png":{"compressionLevel":9}}". The "compressionLevel": 9 is specifying the level of compression for PNG images. Higher numbers here indicate greater compression, leading to smaller file sizes at the cost of potentially increased processing time and possible loss of image quality.
```
Comment (pr-Mais) [inline storage-resize-images/README.md:97]:
```suggestion
* Output options for selected formats: Provide an optional output option stringified object containing Sharp Output Options for selected image types conversion. eg. `{"jpeg": { "quality": 5, "chromaSubsampling": "4:4:4" }, "png": { "palette": true }}` and `{"png":{"compressionLevel":9}}`. The `"compressionLevel": 9` specifies the level of compression for PNG images. Higher numbers here indicate greater compression, leading to smaller file sizes at the cost of potentially increased processing time and possible loss of image quality. 
```

**Review by pr-Mais — CHANGES_REQUESTED** [review]

File: storage-resize-images/extension.yaml [hunk 2]
```diff
@@ -237,8 +237,8 @@ params:
   - param: OUTPUT_OPTIONS
     label: Output options for selected formats
     description: >
-      Provide a optional output option stringified object containing Sharp Output Options for selected image types conversion.
-      eg. `{"jpeg": { "quality": 5, "chromaSubsampling": "4:4:4" }, "png": { "pallete": true }}`
+      Provide an optional output option stringified object containing Sharp Output Options for selected image types conversion. eg. `{"jpeg": { "quality": 5, "chromaSubsampling": "4:4:4" }, "png": { "palette": true }}` and `{"png":{"compressionLevel":9}}`. The `"compressionLevel": 9` specifies the level of compression for PNG images. Higher numbers here indicate greater compression, leading to smaller file sizes at the cost of potentially increased processing time and possible loss of image quality.
```
Comment (pr-Mais) [inline storage-resize-images/extension.yaml:240]:
```suggestion
      Provide an optional output option as a stringified object containing Sharp Output Options for selected image types conversion. eg. `{"jpeg": { "quality": 5, "chromaSubsampling": "4:4:4" }, "png": { "palette": true }}` and `{"png":{"compressionLevel":9}}`. The `"compressionLevel": 9` specifies the level of compression for PNG images. Higher numbers here indicate greater compression, leading to smaller file sizes at the cost of potentially increased processing time and possible loss of image quality.  
```

**Review by pr-Mais — CHANGES_REQUESTED** [review]

**Comment by pr-Mais** [comment]
Can you please add more explanation on how to use `fit` in the options? To resolve this issue #1891

File: storage-resize-images/extension.yaml [hunk 3]
```diff
@@ -247,8 +247,7 @@ params:
   - param: SHARP_OPTIONS
     label: Sharp constructor options for resizing images
     description: >
-      Provide a optional output option stringified object containing Sharp Output Options for selected image types conversion.
-      eg. `{ failOnError: false, limitInputPixels: true }`
+      Provide a optional output option stringified object containing Sharp Output Options for selected image types conversion. eg. `{ failOnError: false, limitInputPixels: true, fit: 'inside' }` The`fit: 'inside'` option ensures image fits within given dimensions, maintaining aspect ratio, scaling down as needed without cropping or distortion. Learn more about [`Sharp constructor options`](https://sharp.pixelplumbing.com/api-resize#resize).
```
Comment (pr-Mais) [inline storage-resize-images/extension.yaml:250]:
```suggestion
      Provide an optional output option stringified object containing Sharp Output Options for selected image types conversion. eg. `{ "failOnError: false, "limitInputPixels": true, "fit: "inside" }` The`"fit": "inside"` option ensures the image fits within given dimensions, maintaining aspect ratio, scaling down as needed without cropping or distortion. Learn more about [`Sharp constructor options`](https://sharp.pixelplumbing.com/api-resize#resize).
```

Comment (pr-Mais) [reply storage-resize-images/extension.yaml:250]: It should be a JSON string hence keys must be inside quotes.

**Review by pr-Mais — CHANGES_REQUESTED** [review]

**Comment by pr-Mais** [comment]
@jauntybrain do you think we can add a validation regex for JSON strings?

**Comment by jauntybrain** [comment]
> @jauntybrain do you think we can add a validation regex for JSON strings?

I doubt it. We would need complex nested validation to validate JSON, which we cannot do with `extension.yaml`.
This also might cause new edge cases -  I would not risk it

File: storage-resize-images/extension.yaml [hunk 4]
```diff
@@ -247,8 +247,7 @@ params:
   - param: SHARP_OPTIONS
     label: Sharp constructor options for resizing images
     description: >
-      Provide a optional output option stringified object containing Sharp Output Options for selected image types conversion.
-      eg. `{ failOnError: false, limitInputPixels: true }`
+      Provide an optional output option stringified object containing Sharp Output Options for selected image types conversion. eg. `{ "failOnError": false, "limitInputPixels": true, "fit": "inside" }` The `"fit": "inside"` option ensures the image fits within given dimensions, maintaining aspect ratio, scaling down as needed without cropping or distortion. Learn more about [`Sharp constructor options`](https://sharp.pixelplumbing.com/api-resize#resize).
```
Comment (pr-Mais) [inline storage-resize-images/extension.yaml:250]: The description here is inaccurate, it's not "output options", try to make it more clear

**Review by pr-Mais — CHANGES_REQUESTED** [review]

**Comment by pr-Mais** [comment]
@jauntybrain can you resolve the conflict

**Review by pr-Mais — APPROVED** [review]

**Pull request merged**